### PR TITLE
[pt][quant] avgpool3d for graph mode

### PR DESCRIPTION
**Pull request by dskhudia**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#36598 [pt][quant] avgpool3d for graph mode**

avgpool3d op for graph mode quantization

Differential Revision: [D21023035](https://our.internmc.facebook.com/intern/diff/D21023035/)

### CODE REVIEW

File: test/quantization/test_quantize_script.py [hunk 1]
```diff
@@ -1398,7 +1398,9 @@ class M(torch.nn.Module):
             def __init__(self):
                 super(M, self).__init__()
                 self.maxpool = torch.nn.MaxPool2d(kernel_size=3)
-                self.adaptive_avgpool = torch.nn.AdaptiveAvgPool2d((1, 1))
+                self.adaptive_avgpool1d = torch.nn.AdaptiveAvgPool1d((1))
+                self.adaptive_avgpool2d = torch.nn.AdaptiveAvgPool2d((1, 1))
+                self.adaptive_avgpool3d = torch.nn.AdaptiveAvgPool3d((1, 1, 1))
                 self.dropout = torch.nn.Dropout()
                 self.avgpool = torch.nn.AvgPool2d(3)
```
Comment (jerryzh168) [inline test/quantization/test_quantize_script.py:1405]: looks good, please add `avgpool1d` module as well

File: test/quantization/test_quantize_script.py [hunk 2]
```diff
@@ -1398,7 +1398,9 @@ class M(torch.nn.Module):
             def __init__(self):
                 super(M, self).__init__()
                 self.maxpool = torch.nn.MaxPool2d(kernel_size=3)
```
Comment (jerryzh168) [inline test/quantization/test_quantize_script.py:1400]: and maxpool1d and maxpool3d module

**Comment by jerryzh168** [comment]
basically make sure we have {avg_pool, max_pool, adapative_avg_pool} * {1d, 2d, 3d} * {module, functional}

**Comment by jerryzh168** [comment]
looks like functional maxpool is internal, so we don't need that. and there will be 3 * 3 * 2 - 3 * 3 = 9 different entries for this

File: test/quantization/test_quantize_script.py [hunk 3]
```diff
@@ -1397,18 +1397,36 @@ def test_swap_dequantize_all_ops(self):
         class M(torch.nn.Module):
             def __init__(self):
                 super(M, self).__init__()
-                self.maxpool = torch.nn.MaxPool2d(kernel_size=3)
-                self.adaptive_avgpool = torch.nn.AdaptiveAvgPool2d((1, 1))
+                self.maxpool1d = torch.nn.MaxPool2d(kernel_size=3)
+                self.maxpool2d = torch.nn.MaxPool2d(kernel_size=3)
+                self.maxpool3d = torch.nn.MaxPool2d(kernel_size=3)
```
Comment (jerryzh168) [inline test/quantization/test_quantize_script.py:1402]: MaxPool1d and MaxPool3d are missing

File: torch/csrc/jit/passes/quantization.cpp [hunk 4]
```diff
@@ -69,8 +69,14 @@ std::vector<std::string> _quantizable_aten_funcs = {
 // so we propagate observed property from %input_tensor to the
 // output of the `prim::CallFunction`
 std::vector<std::string> _single_input_general_call_funcs = {
+    "adaptive_avg_pool1d",
     "adaptive_avg_pool2d",
+    "adaptive_avg_pool3d",
+    "avg_pool1d",
+    "avg_pool2d",
+    "avg_pool3d",
     "_max_pool2d",
```
Comment (jerryzh168) [inline torch/csrc/jit/passes/quantization.cpp:78]: do we neeed `_max_pool1d` here?

File: test/quantization/test_quantize_script.py (hunk 3 above)
Comment (dskhudia) [reply test/quantization/test_quantize_script.py:1402]: Sorry. Typo.

**Review by jerryzh168 — APPROVED** [review]
Looks good, thanks!

**Pull request closed without merging**